### PR TITLE
ManageWikiNamespaces: Add script to add namespaces

### PR DESCRIPTION
**Pull request by paladox**

### CODE REVIEW

File: maintenance/addNamespaces.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    $this->addOption( 'nsadditional', 'Wheather the namespace provides additional things.', true, true );
+
+    $this->db = wfGetDB( DB_MASTER, [], $wgCreateWikiDatabase );
+	}
+
+	public function execute() {
+		global $wgDBname;
+
+		if ( !ManageWiki::checkSetup( 'namespaces' ) ) {
+			$this->fatalError( 'ManageWiki Namespaces is disabled on this wiki.' );
```
Comment (JohnFLewis) [inline maintenance/addNamespaces.php]: Is this an issue?

File: maintenance/addNamespaces.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		}
+
+    $this->insertNamespace();
+	}
+	
+	public function insertNamespace() {
+		global $wgDBname;
+
+		$this->db->insert(
```
Comment (JohnFLewis) [inline maintenance/addNamespaces.php]: Curious why we can't use the ManageWikiNamespace helper functions?

File: maintenance/addNamespaces.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+			'mw_namespaces',
+			[
+				'ns_dbname' => $this->getOption( 'nsdb' ) ? (string)$this->getOption( 'nsdb' ) : $wgDBname,
```
Comment (JohnFLewis) [inline maintenance/addNamespaces.php]: Should absolutely always be ran on the intended wiki.

File: maintenance/addNamespaces.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		$this->addOption( 'nsid', 'The id for the namespace, e.g 1.', true, true );
+    $this->addOption( 'nsname', 'Name of the namespace, e.g Module.', true, true );
+    $this->addOption( 'nssearchable', 'Wheather the namespace is searchable.', true, true );
+		$this->addOption( 'nssubpages', 'Wheather the namespace has a subpage.', true, true );
+		$this->addOption( 'nscontent', 'Wheather the namespace has content', true, true );
+    $this->addOption( 'nscontentmodel', 'Content model for the namespace.', true, true );
+    $this->addOption( 'nsprotection', 'Wheather this namespace has protection.', true, true );
+    $this->addOption( 'nsaliases', 'Wheather the namespace has an alternative name.', true, true );
+    $this->addOption( 'nscore', 'Wheather the namespace is provided by mediawiki core.', true, true );
+    $this->addOption( 'nsadditional', 'Wheather the namespace provides additional things.', true, true );
```
Comment (JohnFLewis) [inline maintenance/addNamespaces.php]: This is a misunderstanding of what this entire column is for.

File: maintenance/addNamespaces.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+		$this->addOption( 'nsdb', 'Name of the namepsace db.', false, true );
+		$this->addOption( 'nsid', 'The id for the namespace, e.g 1.', true, true );
+    $this->addOption( 'nsname', 'Name of the namespace, e.g Module.', true, true );
+    $this->addOption( 'nssearchable', 'Wheather the namespace is searchable.', true, true );
+		$this->addOption( 'nssubpages', 'Wheather the namespace has a subpage.', true, true );
+		$this->addOption( 'nscontent', 'Wheather the namespace has content', true, true );
+    $this->addOption( 'nscontentmodel', 'Content model for the namespace.', true, true );
+    $this->addOption( 'nsprotection', 'Wheather this namespace has protection.', true, true );
+    $this->addOption( 'nsaliases', 'Wheather the namespace has an alternative name.', true, true );
+    $this->addOption( 'nscore', 'Wheather the namespace is provided by mediawiki core.', true, true );
```
Comment (JohnFLewis) [inline maintenance/addNamespaces.php]: Core isn't about how its added, core is about whether the name can be changed (aka rename).

File: maintenance/addNamespaces.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+				'ns_namespace_id' => (int)$this->getOption( 'nsid' ),
+				'ns_namespace_name' => (string)$this->getOption( 'nsname' ),
+				'ns_searchable' => (int)$this->getOption( 'nssearchable' ),
+				'ns_subpages' => (int)$this->getOption( 'nssubpages' ),
+				'ns_content' => (int)$this->getOption( 'nscontent' ),
+        'ns_content_model' => (string)$this->getOption( 'nscontentmodel' ),
+				'ns_protection' => (string)$this->getOption( 'nsprotection' ),
+				'ns_aliases' => (string)json_encode( $this->getOption( 'nsaliases' ) ),
+				'ns_core' => (int)$this->getOption( 'nscore' ),
+        'ns_additional' => (string)json_encode( $this->getOption( 'nsadditional' ) ),
```
Comment (JohnFLewis) [inline maintenance/addNamespaces.php]: I wouldn't bother dealing with this via a script at all. I don't think inputting a PHP array via bash is smart

File: maintenance/addNamespaces.php [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+			[
+				'ns_dbname' => $this->getOption( 'nsdb' ) ? (string)$this->getOption( 'nsdb' ) : $wgDBname,
+				'ns_namespace_id' => (int)$this->getOption( 'nsid' ),
+				'ns_namespace_name' => (string)$this->getOption( 'nsname' ),
+				'ns_searchable' => (int)$this->getOption( 'nssearchable' ),
+				'ns_subpages' => (int)$this->getOption( 'nssubpages' ),
+				'ns_content' => (int)$this->getOption( 'nscontent' ),
+        'ns_content_model' => (string)$this->getOption( 'nscontentmodel' ),
+				'ns_protection' => (string)$this->getOption( 'nsprotection' ),
+				'ns_aliases' => (string)json_encode( $this->getOption( 'nsaliases' ) ),
```
Comment (JohnFLewis) [inline maintenance/addNamespaces.php]: Same as ns_additional

File: maintenance/addNamespaces.php (hunk 3 above)
Comment (paladox) [reply maintenance/addNamespaces.php]: I want to support setting it for 'default'. So this flag should only be used if you intent to add it to the 'default' db name.

File: maintenance/addNamespaces.php (hunk 4 above)
Comment (paladox) [reply maintenance/addNamespaces.php]: What should i change this too?

File: maintenance/addNamespaces.php (hunk 6 above)
Comment (paladox) [reply maintenance/addNamespaces.php]: Ok.

File: maintenance/addNamespaces.php (hunk 4 above)
Comment (JohnFLewis) [reply maintenance/addNamespaces.php]: See below comments

File: maintenance/addNamespaces.php (hunk 3 above)
Comment (JohnFLewis) [reply maintenance/addNamespaces.php]: So then this isn't a "what database flag" it's a "is this a default namespace" flag

File: maintenance/addNamespaces.php (hunk 2 above)
Comment (paladox) [reply maintenance/addNamespaces.php]: I was just trying that :)

File: maintenance/addNamespaces.php (hunk 1 above)
Comment (paladox) [reply maintenance/addNamespaces.php]: Not really i guess.

**Pull request merged**